nixos/modules/services/monitoring/netdata.nix

```diff
@@ -0,0 +1,80 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.netdata;
+
+  configFile = pkgs.writeText "netdata.conf" cfg.configText;
+
+in {
+  options = {
+    services.netdata = {
+      enable = mkEnableOption "netdata";
+      user = mkOption {
+        type = types.str;
+        default = "netdata";
+        description = "User account under which netdata runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "netdata";
+        description = "Group under which netdata runs.";
+      };
+
+      configText = mkOption {
+        type = types.nullOr types.str;
```

Comment:
This could be made `type = types.line;` with `""` as default. Removing `nullOr` save the need for `if isNull cfg.configText then "" else "` later, and `types.lines` will provide better value merging in case the option is defined multiple times.